### PR TITLE
ORCHESTRATIONS: Persistance - PurgeCsvRecords

### PR DESCRIPTION
**Pull request by JParkesAD**
CLOSES #AB17124

### CODE REVIEW

File: ISL.ReIdentification.WebApps.Portals/ISL.ReIdentification.Portals.Server/Program.cs [hunk 1]
```diff
@@ -212,8 +213,13 @@ private static void AddProcessingServices(IServiceCollection services)
             services.AddTransient<IUserAccessProcessingService, UserAccessProcessingService>();
         }
 
-        private static void AddOrchestrationServices(IServiceCollection services)
+        private static void AddOrchestrationServices(IServiceCollection services, IConfiguration configuration)
         {
+            CsvReIdentificationConfigurations csvReIdentificationConfigurations = configuration
+                .GetSection("csvReIdentificationConfigurations")
+                    .Get<CsvReIdentificationConfigurations>();
+
```
Comment (cjdutoit) [inline ISL.ReIdentification.WebApps.Portals/ISL.ReIdentification.Portals.Server/Program.cs:221]: Validate that the `csvReIdentificationConfigurations` is not null.
If this is null, you would look at the int initialising to a value of 0 which will mean immediate expiry.  
I think validate that value is greater than 5

File: ISL.ReIdentification.Core/Services/Orchestrations/Persists/PersistanceOrchestrationService.cs [hunk 2]
```diff
@@ -137,5 +155,51 @@ public ValueTask<AccessRequest> RetrieveCsvIdentificationRequestByIdAsync(
 
             return new AccessRequest { CsvIdentificationRequest = csvIdentificationRequest };
         });
+
+        public ValueTask PurgeCsvReIdentificationRecordsThatExpired() =>
+        TryCatch(async () =>
+        {
+            DateTimeOffset dateTimeOffset = await this.dateTimeBroker.GetCurrentDateTimeOffsetAsync();
+
+            DateTimeOffset expiryDate =
```
Comment (cjdutoit) [inline ISL.ReIdentification.Core/Services/Orchestrations/Persists/PersistanceOrchestrationService.cs:164]: Need validation to ensure `this.csvReIdentificationConfigurations` is not null and then that 
`this.csvReIdentificationConfigurations.ExpireAfterMinutes` has a value greater than 5

File: ISL.ReIdentification.Core.Tests.Unit/Services/Orchestrations/Persists/PersistanceOrchestrationServiceTests.Logic.PurgeCsvReIdentificationRecordsThatExpired.cs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+                    service.AddAccessAuditAsync(It.Is(SameAccessAuditAs(accessAudit))))
+                        .ReturnsAsync(accessAudit);
+            }
+
+            // when
+            await this.persistanceOrchestrationService
+                .PurgeCsvReIdentificationRecordsThatExpired();
+
+            // then
+
```
Comment (MichaelOSullivanAnswer) [inline ISL.ReIdentification.Core.Tests.Unit/Services/Orchestrations/Persists/PersistanceOrchestrationServiceTests.Logic.PurgeCsvReIdentificationRecordsThatExpired.cs:77]: remove empty line

**Review by cjdutoit — CHANGES_REQUESTED** [review]

File: ISL.ReIdentification.Core.Tests.Unit/Services/Orchestrations/Persists/PersistanceOrchestrationServiceTests.Logic.PurgeCsvReIdentificationRecordsThatExpired.cs (hunk 3 above)
Comment (JParkesAD) [reply ISL.ReIdentification.Core.Tests.Unit/Services/Orchestrations/Persists/PersistanceOrchestrationServiceTests.Logic.PurgeCsvReIdentificationRecordsThatExpired.cs]: Fixed

File: ISL.ReIdentification.WebApps.Portals/ISL.ReIdentification.Portals.Server/Program.cs (hunk 1 above)
Comment (JParkesAD) [reply ISL.ReIdentification.WebApps.Portals/ISL.ReIdentification.Portals.Server/Program.cs:221]: Change pushed

File: ISL.ReIdentification.Core/Services/Orchestrations/Persists/PersistanceOrchestrationService.cs (hunk 2 above)
Comment (JParkesAD) [reply ISL.ReIdentification.Core/Services/Orchestrations/Persists/PersistanceOrchestrationService.cs:166]: Change pushed

File: ISL.ReIdentification.Core/Services/Orchestrations/Persists/PersistanceOrchestrationService.Exceptions.cs [hunk 4]
```diff
@@ -104,6 +106,72 @@ private async ValueTask<AccessRequest> TryCatch(ReturningAccessRequestFunction r
...
 
+        private async ValueTask TryCatch(ReturningNothingFunction returningNothingFunction)
+        {
+            try
+            {
+                await returningNothingFunction();
+            }
+            catch (NullCsvReIdentificationConfigurationException nullCsvReIdentificationConfigurationException)
+            {
+                throw await CreateAndLogValidationExceptionAsync(nullCsvReIdentificationConfigurationException);
```
[the remark this answers is not in the document]
Comment (JParkesAD) [reply ISL.ReIdentification.Core/Services/Orchestrations/Persists/PersistanceOrchestrationService.Exceptions.cs]: Change pushed

**Review by cjdutoit — APPROVED** [review]

**Review by cjdutoit — APPROVED** [review]

**Pull request merged**